File: leetcode/1961-check-if-string-is-a-prefix-of-array/CheckIfStringIsAPrefixOfArray.ts
function isPrefixString(s: string, words: string[]): boolean {
    let j = 0;

    for (let i = 0; i < words.length && j < s.length; i++) {
        const left = words[i];
        const right = s.substring(j, j + left.length);

        if (left != right) {
            return false;
        }

        j = j + left.length;
    }

    return j == s.length;
}

const s = "iloveleetcode";
const words = ["i", "love", "leetcode", "apples"];

console.log(isPrefixString(s, words));

File: leetcode/1001-1999/1961-check-if-string-is-a-prefix-of-array/CheckIfStringIsAPrefixOfArray.java
public class CheckIfStringIsAPrefixOfArray {
  public static void main(String[] args) {
    String s = "iloveleetcode";
    String[] words = { "i", "love", "leetcode", "apples" };

    System.out.println(new Solution().isPrefixString(s, words));
  }
}

class Solution {
  public boolean isPrefixString(String s, String[] words) {
    int j = 0;

    for (int i = 0; i < words.length && j < s.length(); i++) {
      String left = words[i];

      if (s.length() < j + left.length()) {
        return false;
      }

      String right = s.substring(j, j + left.length());

      if (!left.equals(right)) {
        return false;
      }

      j = j + left.length();
    }

    return j == s.length();
  }
}
